x-pack/legacy/plugins/ml/public/data_frame_analytics/pages/analytics_management/components/analytics_list/action_delete.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment, SFC, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiButtonEmpty,
+  EuiConfirmModal,
+  EuiOverlayMask,
+  EuiToolTip,
+  EUI_MODAL_CONFIRM_BUTTON,
+} from '@elastic/eui';
+
+import { deleteAnalytics } from '../../services/analytics_service';
+
+import {
+  checkPermission,
+  createPermissionFailureMessage,
+} from '../../../../../privilege/check_privilege';
+
+import { DataFrameAnalyticsListRow, DATA_FRAME_TASK_STATE } from './common';
+
+interface DeleteActionProps {
+  item: DataFrameAnalyticsListRow;
+}
+
+export const DeleteAction: SFC<DeleteActionProps> = ({ item }) => {
```

Comment:
we probably should be using `FC` for all new code

Reply:
Changed for all analytics code in aec4524.